commit 819cbb120eaec7e014e5abd029260db1ca8c5735
Author: Vasiliy Kulikov <segoon@openwall.com>
Date:   Sun Jun 26 12:56:22 2011 +0400

    staging: comedi: fix infoleak to userspace
    
    driver_name and board_name are pointers to strings, not buffers of size
    COMEDI_NAMELEN.  Copying COMEDI_NAMELEN bytes of a string containing
    less than COMEDI_NAMELEN-1 bytes would leak some unrelated bytes.
    
    Signed-off-by: Vasiliy Kulikov <segoon@openwall.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/drivers/staging/comedi/comedi_fops.c b/drivers/staging/comedi/comedi_fops.c
index 15a209f..419976b 100644
--- a/drivers/staging/comedi/comedi_fops.c
+++ b/drivers/staging/comedi/comedi_fops.c
@@ -383,8 +383,8 @@ static int do_devinfo_ioctl(struct comedi_device *dev,
 	/* fill devinfo structure */
 	devinfo.version_code = COMEDI_VERSION_CODE;
 	devinfo.n_subdevs = dev->n_subdevices;
-	memcpy(devinfo.driver_name, dev->driver->driver_name, COMEDI_NAMELEN);
-	memcpy(devinfo.board_name, dev->board_name, COMEDI_NAMELEN);
+	strlcpy(devinfo.driver_name, dev->driver->driver_name, COMEDI_NAMELEN);
+	strlcpy(devinfo.board_name, dev->board_name, COMEDI_NAMELEN);
 
 	if (read_subdev)
 		devinfo.read_subdevice = read_subdev - dev->subdevices;
